src/utils/common.ts

```diff
@@ -260,13 +262,16 @@ export async function estimateTotalGasRequiredByUnsignedTransaction(
   const voidSigner = new VoidSigner(senderAddress, provider);
 
   // Estimate the Gas units required to submit this transaction.
-  let nativeGasCost = await voidSigner.estimateGas(unsignedTx);
+  let nativeGasCost = await voidSigner.estimateGas({
+    ...unsignedTx,
+    gasPrice,
+    gasLimit,
+  });
   let tokenGasCost: BigNumber;
 
   // OP stack is a special case; gas cost is computed by the SDK, without having to query price.
   if (chainIsOPStack(chainId)) {
     assert(isOptimismL2Provider(provider), `Unexpected provider for chain ID ${chainId}.`);
-    assert(gasPrice === undefined, `Gas price (${gasPrice}) supplied for Optimism gas estimation (unused).`);
```

Comment:
Not sure how to handle this, since the gas price is never used for the OP stack deployments... If the caller supplies a gas price and we return a result, I think it's reasonable for them to expect that their gas price was actually used.

Reply:
That's a good point. I added a commit here https://github.com/across-protocol/sdk/pull/680/commits/87575e72ba58f400e52b6ecae06a0d8b958b57d9 that passes on the gas price if provided. This even saves some RPC calls which is also nice.